clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -635,6 +631,11 @@ public void transitionToFatal() {
         });
     }
 
+    // Visible for testing
+    String memberIdInfoForLog() {
+        return memberId.isEmpty() ? "without ID" : memberId;
```

Comment:
in the protocol, the memberId is nullable. maybe it should handle the null too

Reply:
+1
Don’t know why in the request protocol we have an empty string as the default, but there is no such guarantee in the response protocol. However, since this is the case, we should handle null to avoid NPE or weird logs.

Reply:
Agreed, added the extra check to play safe. On the client side it defaults to empty string, but at the protocol level it's indeed nullable (ex. we could end up with responses without member ID in cases of errors on the first HB to join)